src/test/scala/org/camunda/feel/impl/interpreter/InterpreterExpressionTest.scala

```diff
@@ -264,22 +264,22 @@ class InterpreterExpressionTest
 
   "A variable name" should "not be a key-word" in {
 
-    eval("some = true") shouldBe a[ValError]
-    eval("every = true") shouldBe a[ValError]
-    eval("if = true") shouldBe a[ValError]
-    eval("then = true") shouldBe a[ValError]
-    eval("else = true") shouldBe a[ValError]
-    eval("function = true") shouldBe a[ValError]
-    eval("for = true") shouldBe a[ValError]
-    eval("between = true") shouldBe a[ValError]
-    eval("instance = true") shouldBe a[ValError]
-    eval("of = true") shouldBe a[ValError]
-    eval("not = true") shouldBe a[ValError]
-    eval("in = true") shouldBe a[ValError]
-    eval("satisfies = true") shouldBe a[ValError]
-    eval("and = true") shouldBe a[ValError]
-    eval("or = true") shouldBe a[ValError]
-    eval("return = true") shouldBe a[ValError]
+    eval("some = true", Map("some" -> 1)) shouldBe a[ValError]
+    eval("every = true", Map("every" -> 1)) shouldBe a[ValError]
+    eval("if = true", Map("if" -> 1)) shouldBe a[ValError]
+    eval("then = true", Map("then" -> 1)) shouldBe a[ValError]
+    eval("else = true", Map("else" -> 1)) shouldBe a[ValError]
+    eval("function = true", Map("function" -> 1)) shouldBe a[ValError]
+    eval("for = true", Map("for" -> 1)) shouldBe a[ValError]
+    eval("between = true", Map("between" -> 1)) shouldBe a[ValError]
+    eval("instance = true", Map("instance" -> 1)) shouldBe a[ValError]
+    eval("of = true", Map("of" -> 1)) shouldBe a[ValError]
+    eval("not = true", Map("not" -> 1)) shouldBe a[ValError]
+    eval("in = true", Map("in" -> 1)) shouldBe a[ValError]
+    eval("satisfies = true", Map("satisfies" -> 1)) shouldBe a[ValError]
+    eval("and = true", Map("and" -> 1)) shouldBe a[ValError]
+    eval("or = true", Map("or" -> 1)) shouldBe a[ValError]
+    eval("return = true", Map("return" -> 1)) shouldBe a[ValError]
```

Comment:
@saig0 I'm not quire sure about these tests. I think they weren't good tests to begin with. I'd expect an expression using a keyword as variable would fail during parsing, but apparently that's not the case.

From my limit understanding before this would also return a `ValError`, but it must've been an error about how the variable didn't exist.

The way it is not I make sure the variable is defined. This still returns a `ValError` as it the name is a keyword. It's a bit odd though, we don't care about it being a keyword, unless it's actually defined. If it's not defined we will consider it to be valid and consider it to be a `ValNull`.

Reply:
I see. You're right. :+1:  The test case is partially broken. Some but not all expressions fail because of the parsing. The parser doesn't list all of the keywords as reserved words. 

The intention of the test is to verify that "a variable name should not be a keyword". The changes don't solve the issue.

I suggest making the expected error more clear. For example, by using a custom [matcher](https://www.scalatest.org/scaladoc/3.2.16/org/scalatest/matchers/BeMatcher.html):

```
  "A variable name" should "not be a key-word" in {
    eval("else") shouldBe aParseError
    // ...
  }

  class EvaluationErrorMatcher(expectedMessage: String) extends BeMatcher[Val] {
    override def apply(result: Val): MatchResult =
      result match {
        case ValError(failure) => MatchResult(
          failure.startsWith(expectedMessage),
          s"$result doesn't start with '$expectedMessage'",
          s"$result starts with '$expectedMessage'",
        )
        case _ => MatchResult(
          false,
          s"$result is not an error",
          s"$result is an error"
        )
      }
  }

  def anError(expectedMessage: String) = new EvaluationErrorMatcher(expectedMessage)
  def aParseError(): EvaluationErrorMatcher = anError(expectedMessage = "failed to parse expression")
```

And we should ignore the cases that are allowed (for now). 

---

:fast_forward: Feel free to fix this test case in a separate PR.

Reply:
https://github.com/camunda/feel-scala/pull/680 🙂